README.md

```diff
@@ -45,25 +44,86 @@ Create an input configuration object and add a redux-inputs reducer to your redu
-
-    inputConfig = {
-        _form: {
-            reduxMountPoint: 'loginModal.userInputs'
-        }
-        ... other inputs config ...
-    }
+### Step 4 - Connect
+
+    import { connectWithInputs } from 'redux-inputs';
+    
+    const connectWithMyForm = connectWithInputs(inputConfig);
+    
+    YourForm = connectWithMyForm(mapStateToProps, mapDispatchToProps, mergeProps, options)(YourForm);
+    
+Create a connect function for your form, which takes the `inputConfig` and creates a function that has the same 
+interface as [react-redux](https://github.com/reactjs/react-redux)'s `connect`. This will pass down these additional props to your component, in addition to any 
+other redux state specified:
+
+#### Props passed down to component
+
+- `inputProps` *{Object}* contains props for each input in the inputConfig
+
+- `inputActions` *{Object}*
+    - `inputActions.setInputs(newState, meta)`
+    - `inputActions.updateAndValidate(changes, meta)`
+    - `inputActions.validateInputs(inputKeys, meta)`
+    - `inputActions.resetInputs(inputKeys, meta)`
+    - `inputActions.initializeInputs(changes, meta)`
+    
+- `form` *{Object}*
+    - `form.values` *{Object}* key-value pairs of the values of all inputs
+    - `form.validating` *{Boolean}* true if any inputs are currently async validating
+    - `form.pristine` *{Boolean}* true if all inputs are untouched, reset, or initialized.
+    
+
+### Step 5 - Wrapper
+
+    import { ReduxInputsWrapper } from 'redux-inputs';
+    
+    let Input = ({id, value, error, errorText, onChange}) => (
+        <div>
+            <input name={id} onChange={(e) => onChange(e.target.value)}/>
+            {error ? <div>{errorText}</div> : null} 
+        </div>
+    );
+    Input = ReduxInputsWrapper(Input);
+
+Higher order component that wraps input components and wires them up to the state. This allows all types of inputs to 
+conform to the same API, and be easily understood and swapped out.
 
-In this example your state would look like this:
+#### Arguments
+- `WrappedComponent` *(Component)*
+- `options` *(Object)*
+- - `onChangeTransform` *(Function)* [optional] Turn a browser change event into a value. (e.g. `e => e.target.value` for `input`)
 
-    {
-        loginModal: {
-            userInputs: { ... }
-        }
+#### Props that must be implemented in wrapped component
+
+- id *(string)*: unique id for this input, can be used to link label and input
+- value: represents the current value as defined by the input's state
+- error *(boolean)*: True if the current value is invalid
+- errorText *(string)*
+- onChange *(function)*: Takes one parameter - the new value
+
+#### Props added to composed component
+
+- parser *(function)* transforms the new value before validating and updating
+- formatter *(function)* transforms the stored value before passing to the component
+- resolve *(function)* callback after successful change - same as resolve of updateAndValidate
```

Comment:
Rename resolve/reject to something more descriptive e.g. onValidationSuccess/onValidationError and fix callback api to call them with the same (input, inputs, state, dispatch) as all the other functions